src/merkle/merkle_tree.rs

```diff
@@ -34,10 +29,12 @@ impl MerkleTree {
 
         // create un-initialized vector to hold all tree nodes
         let mut nodes = unsafe { uninit_vector(2 * n) };
-        nodes[0] = [Felt::ZERO; 4];
+        nodes[0] = RpoDigest::default();
 
         // copy leaves into the second part of the nodes vector
-        nodes[n..].copy_from_slice(&leaves);
+        nodes[n..].iter_mut().zip(leaves).for_each(|(node, leaf)| {
+            *node = RpoDigest::from(leaf);
+        });
```

Comment:
Do you know if this affected performance in noticeable way?

Reply:
Thank you! I would have probably tested with bigger trees - but I think that's fine for now.

Reply:
No change with $2^12$ leaves either.

Reply:
Thank you for checking!